### DISCUSSION TITLE
AddDivisionEquality constraint for jssp

### DISCUSSION THREAD

**Discussion by ensarovski** [google/or-tools#3249] [CP-SAT questions] [answered]
I need this constraint, but it gives an error like that 'calling // on a linear expression is not supported, please use CpModel.AddDivisionEquality'
   if  (task_id>1):
        model.Add(starts[(job_id, task_id)] >= starts[(job_id, task_id-1)]*11/10)

**Comment by lperron** [comment]
var * 11 is an LinearExpression, (var * 11) / 10 is a division. You need to create an intemediate variables

```
tmp_var = model.NewIntVar(...)
model.AddDivisionEquality(tmp_var, start * 11, 10)
model.Add(new_start >= tmp_var)
```

this being said, adding a slack proportional to the start time seems a bad idea. What happens if you rescale `time 0` to be far in the past?

**Comment by vaidzelevich** [comment, marked as answer]
Hello. In this particular case, you can multiply both sides of the inequality by the denominator:

```python
model.Add(10 * starts[(job_id, task_id)] >= 11 * starts[(job_id, task_id-1)])
```

**Comment by lperron** [comment]
Yes

Le dim. 17 avr. 2022, 12:44, vaidzelevich ***@***.***> a
écrit :

**Comment by ensarovski** [comment]
Thank you so much @vaidzelevich and @lperron
